python/ray/ml/session.py

```diff
@@ -0,0 +1,98 @@
+from typing import Optional
+
+import logging
+import os
+import shutil
+
+from ray import tune
+from ray import train
+from ray.ml.checkpoint import Checkpoint
+from ray.tune.function_runner import StatusReporter
+
+logger = logging.getLogger(__name__)
+
+_LOCAL_PATH = "local_path"
+_DATA_DICT = "data_dict"
+
+
+def _get_session():
+    # TODO(xwjiang): Currently the ml session only multiplexes between Tune and Train session.
+    #  We should extract common functionalities if possible.
+    from ray.tune.session import _session as tune_session
+    from ray.train.session import _session as train_session
+    if train_session and tune_session:
+        logger.warning("Expected to be either in tune session or train session but not both.")
+        return None
+    if not (train_session or tune_session):
+        logger.warning("In neither tune session nor train session!")
+        return None
+    return train_session or tune_session
+
+
+def save(epoch: int, checkpoint: Optional[Checkpoint] = None, metrics: Optional[dict] = None):
```

Comment:
Let's have a chat with some people about APIs here! I think we should usually use `save_checkpoint` if we're saving checkpoints here, except if we use the API to save other things as well, like artifacts.

Another option would be to consolidate `session.report` with this - e.g. `session.report(some_metric=a, checkpoint=cp)` or so.

I think a mini design doc would be great here!